ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/QuotedIdentifier.java

```diff
@@ -27,11 +27,24 @@
  * @since 2.0.0
  */
 public class QuotedIdentifier {
-    private final String name;
+    private final int ASCII_VALUE = 128;
+    private String name;
 
     public QuotedIdentifier(String name) {
+        boolean isUnicode = false;
         Objects.requireNonNull(name);
-        this.name = StringUtils.quoted(name);
+        char[] characters = name.toCharArray();
+        for (char character: characters) {
+            if (character >= ASCII_VALUE) {
+                isUnicode = true;
+                String string = "" + character;
+                this.name = StringUtils.quoted(name)
```

Comment:
I think there is a problem here, `this.name` gets updated in the loop using `name`, so changes done in previous iterations are discarded.

I think its better to do this as following, (`convertUnicode` method may need to be changed to accept a character)
```java
    public QuotedIdentifier(String name) {
        Objects.requireNonNull(name);
        StringBuilder convertedName = new StringBuilder();
        for (char character : name.toCharArray()) {
            if (character >= ASCII_VALUE) {
                String replacement = StringUtils.convertUnicode(character);
                convertedName.append(replacement);
            } else {
                convertedName.append(character);
            }
        }
        this.name = StringUtils.quoted(convertedName.toString());
    }
```

Reply:
Updated in 3083628406bc711786e3b472071f89c46d21e09e